### DISCUSSION TITLE
Release/5.2.2

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1345] [Announcement]
We now test against, and package with, Python 3.9.8.

As a consequence of this we no longer support Windows 7.
This is due to Python 3.9.x itself not supporting Windows 7.
The application (both EDMarketConnector.exe and EDMC.exe) will crash on startup due to a missing DLL.

This should have no other impact on users or plugin developers, other than the latter now being free to use features that were introduced since the Python 3.7 series.

Developers can check the contents of the `.python-version` file in the source (it's not distributed with the Windows installer) for the currently used version in a given branch.

---

Release 5.2.2
===

This release adds one new feature and addresses some bugs.  We've also updated to using Python 3.9.8.

* Windows now has "minimize to system tray" support.

    - The system tray icon will always be present.  - There is a new option on the Settings > Appearance tab -
     `Minimize to system tray`.
    - When this new option is active, minimizing the application will *also*
      hide the taskbar icon.
    - When the new option is not active, the application will minimize to the
      taskbar as normal.

Bug Fixes
---

* If a CAPI query failed in such a way that no `requests.Response` object
  was made available we attempted to blindly dump the non-existent object.  We now check that it actually exists, and log the specifics of the exception.

* A user experienced the game writing a NavRoute.json file without a
  `Route` array, which caused the application to attempt sending a badly formed `navroute` message to EDDN.  That message was then remembered and constantly retried.

    - We now sanity check the NavRoute.json contents to be sure there *is* a
      `Route` array, even if it is empty.  If it's not present no attempt to send the EDDN message will be made.

      If this scenario occurs the user will see a status line message `No 'Route' array in NavRoute.json contents`.

    - For any EDDN message that receives a 400 status back we will drop it
      from the replay log.

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.2.2'>Release/5.2.2</a>.</em>

**Comment by camy252** [comment]
Anyone else getting the error: 'CAPI query aborted: Cmdr name unknown'.  Was working before so its not my API links and i've tried a couple of fresh installs.

**Comment by Athanasius** [comment]
> Anyone else getting the error: 'CAPI query aborted: Cmdr name unknown'. Was working before so its not my API links and i've tried a couple of fresh installs.

Are you:

1. Not logged fully into the game.
2. Pressing the 'Update' button.

I ask because that message is when EDMC doesn't yet know your commander name (necessary in order to know *which* Commander to query the CAPI for, as we have multi-account support).

**Comment by Athanasius** [comment]
Locking this as https://github.com/EDCD/EDMarketConnector/releases/tag/Release%2F5.2.3 is now available.
